test/integration/puppetlabs/general_puppet/general_puppet_int_test.clj

```diff
@@ -71,3 +72,30 @@
         (is (nil? (get catalog "code_id")))
         (is (logged? #"Non-zero exit code returned while running" :error))
         (is (logged? #"Executed an external process which logged to STDERR: production" :warn)))))))
+
+(deftest ^:integration static-file-content-endpoint-test
+  (logging/with-test-logging
+    (testing "the /static_file_content endpoint successfully streams file content"
+      (bootstrap/with-puppetserver-running
+        app
+        {:jruby-puppet
+         {:max-active-instances num-jrubies}
+         :versioned-code
+         {:code-content-command (script-path "echo")}}
+        (let [response (http-client/get "https://localhost:8140/puppet/v3/static_file_content/foo/bar/?code-id=foobar&environment=test"
+                                        (assoc testutils/ssl-request-options
+                                          :as :stream))]
+          (is (= 200 (:status response)))
+          (is (= "test foobar foo/bar/\n" (slurp (:body response)))))))
+
+    (testing "the /static_file_content endpoint errors if :code-content-command is not set"
+      (bootstrap/with-puppetserver-running
+        app
+        {:jruby-puppet
+         {:max-active-instances num-jrubies}
+         :versioned-code
+         {:code-content-command nil}}
+        (let [response (http-client/get "https://localhost:8140/puppet/v3/static_file_content/foo/bar/?code-id=foobar&environment=test"
+                                        (assoc testutils/ssl-request-options
+                                          :as :stream))]
+          (is (= 500 (:status response))))))))
```

Comment:
Okay, that's reasonable.  I think it'd be good for that to be tested here as well.  And while, you're at it, what do you think about changing that error message in `get-code-content` to something a little better like:

"Cannot retrieve code content because the "versioned-code.code-content-command" setting is not present in configuration."